### PR TITLE
Add Adafruit Trinkey QT2040 board

### PR DESCRIPTION
**Pull request by mcauser**
https://www.adafruit.com/product/5056
https://learn.adafruit.com/adafruit-trinkey-qt2040/pinouts

### CODE REVIEW

File: src/boards/include/boards/adafruit_trinkey_qt2040.h [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+#define PICO_DEFAULT_UART 0
+#endif
+
+#ifndef PICO_DEFAULT_UART_TX_PIN
+#define PICO_DEFAULT_UART_TX_PIN 0
+#endif
+
+#ifndef PICO_DEFAULT_UART_RX_PIN
+#define PICO_DEFAULT_UART_RX_PIN 1
+#endif
```
Comment (lurch) [inline src/boards/include/boards/adafruit_trinkey_qt2040.h:34]: The Trinkey board doesn't seem to break out GP0 or GP1 ?

File: src/boards/include/boards/adafruit_trinkey_qt2040.h [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+#ifndef _BOARDS_ADAFRUIT_TRINKEY_QT2040_H
+#define _BOARDS_ADAFRUIT_TRINKEY_QT2040_H
+
+// For board detection
+#define ADAFRUIT_TRINKEY_QT2040
+
+// On some samples, the xosc can take longer to stabilize than is usual
+#ifndef PICO_XOSC_STARTUP_DELAY_MULTIPLIER
+#define PICO_XOSC_STARTUP_DELAY_MULTIPLIER 64
+#endif
```
Comment (lurch) [inline src/boards/include/boards/adafruit_trinkey_qt2040.h:21]: @dhalbert Is this also needed for this board?

File: src/boards/include/boards/adafruit_trinkey_qt2040.h [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+#define PICO_DEFAULT_I2C_SDA_PIN 16
+#endif
+
+#ifndef PICO_DEFAULT_I2C_SCL_PIN
+#define PICO_DEFAULT_I2C_SCL_PIN 17
+#endif
+
+//------------- SPI -------------//
+#ifndef PICO_DEFAULT_SPI
+#define PICO_DEFAULT_SPI 0
```
Comment (lurch) [inline src/boards/include/boards/adafruit_trinkey_qt2040.h:59]: I don't think you can claim that this board has a "default SPI", given that it looks like no SPI clock pin is available?

File: src/boards/include/boards/adafruit_trinkey_qt2040.h (hunk 2 above)
Comment (dhalbert) [reply src/boards/include/boards/adafruit_trinkey_qt2040.h:21]: It is potentially needed for any current Adafruit board.

File: src/boards/include/boards/adafruit_trinkey_qt2040.h (hunk 3 above)
Comment (mcauser) [reply src/boards/include/boards/adafruit_trinkey_qt2040.h:59]: I wasn't sure if these were needed. Looking at pimoroni_keybow2040.h it seems you can just omit them.
This board doesn't breakout UART or SPI, only I2C over the Stemma connector.
I'll comment out both. Thanks @lurch

File: src/boards/include/boards/adafruit_trinkey_qt2040.h (hunk 1 above)
Comment (mcauser) [reply src/boards/include/boards/adafruit_trinkey_qt2040.h]: Commented out

File: src/boards/include/boards/adafruit_trinkey_qt2040.h (hunk 3 above)
Comment (mcauser) [reply src/boards/include/boards/adafruit_trinkey_qt2040.h]: Commented out

File: src/boards/include/boards/adafruit_trinkey_qt2040.h [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+#endif
+
+//------------- UART -------------//
+// no PICO_DEFAULT_UART
+// no PICO_DEFAULT_UART_TX_PIN
+// no PICO_DEFAULT_UART_RX_PIN
+
+//------------- LED -------------//
+// No normal LED
+// #define PICO_DEFAULT_LED_PIN 13
```
Comment (lurch) [inline src/boards/include/boards/adafruit_trinkey_qt2040.h:30]: Perhaps more consistent (with the UART and SPI sections) if you change this to just `// no PICO_DEFAULT_LED_PIN` ?

**Review by kilograham — APPROVED** [review]

**Pull request merged**